### ISSUE TITLE
Blokada is closed all the time on Samsung S9 when Notifications are set to `No`

### ISSUE THREAD

**Issue by alexanderadam** [blokadaorg/blokada#613]
Blokada ran perfectly fine [since 547 was fixed](https://github.com/blokadaorg/blokada/issues/547) but on some point in the last months (I guess it was December or November) it began being closed every now and then. The notification bar disappears and when I open Blokada it will reestablish the connection again.
It's _not_ the battery optimisation thingy (because I excluded Blokada from battery optimisation).
I don't _think_ that Blokada is killing itself but I guess it's not prohibiting others from killing it. :wink: 

### Steps to reproduce
1. Run Blokada on Samsung S9

### Expected behaviour

Blokada would keep running. :+1: 

### Actual behaviour

Blokada closes. :-1: 

### Configuration
1. :shield: Blokada version: [`4.4.3 (android-28 community release arm64-v8a samsung-starlte touch api compatible)`](https://f-droid.org/en/packages/org.blokada.alarm/#4.4.3)
2. :iphone: Device model: [`Samsung S9`](https://www.samsung.com/ch/smartphones/galaxy-s9/)
3. :robot: Android version / ROM: `9` / `One UI 1.0` / Kernel `4.9.59-16947752`

### Questions
1. *How long have you been using Blokada?*
Since a long time but I'm still struggling why this is relevant. Maybe you want to add a comment to the issue template that explains how and why this is relevant. This would help to answer this question better, I guess.

2. *How long time ago has the problem started to happen?*
Not too long ago. I would say since two months at maximum. But I didn't had a look on it exactly.

3. *How often does this problem happen to you?*
All the time.

### Settings

Notifications: `No`
Start on boot: `Yes`
Settings location: `Internal Storage`
Background animation: `No`
Keep alive: `Yes`
Watchdog: `Yes`
Aggressive wakeup: `Yes`

**PS:** there were similar issues on Blokada [2.2.2](https://github.com/blokadaorg/blokada/issues/73), [3.6](https://github.com/blokadaorg/blokada/issues/317), 3.7 ([1](https://github.com/blokadaorg/blokada/issues/346), [2](https://github.com/blokadaorg/blokada/issues/379), [3](https://github.com/blokadaorg/blokada/issues/498)), [4.3.4](https://github.com/blokadaorg/blokada/issues/553).
**PPS:** Please note that other applications are staying in background perfectly fine on the same device (i.e. messengers like [Conversations](https://github.com/siacs/Conversations) or [Threema](https://threema.ch/de)).

**Comment by Notabas-iC** [comment]
Hello there,
For Samsung devices please either try to enable Always-on-VPN in your system settings or lock Blokada in recents (if it's possible on your device) and see whether that resolves the shutdowns.

**Comment by peterroth** [comment]
and enable "Keep alive" in 'App settings'.
Didn't you update your Android recently?

**Comment by alexanderadam** [comment]
> Turn off "Watchdog".

Wouldn't the watchdog kill messengers as well if it would be faulty? :thinking:

> Didn't you update your Android recently?

No, I didn't but I saw that Blokada stays on when I set Notifications to `Yes`.
However, messengers don't need a permanent notification and they staying on perfectly fine.

**Comment by peterroth** [comment]
> Wouldn't the watchdog kill messengers as well if it would be faulty? 🤔
Can you elaborate on it? I'm afraid I don't get what you mean.

A messenger application also has notifications, if you check, otherwise you won't be notified about new messages. You don't only receive them because somebody wrote you, but because that keeps the service up and running in the background. Notifications increases the process' priority, what will ensure it isn't killed by the system.

**Comment by alexanderadam** [comment]
> A messenger application also has notifications, if you check, otherwise you won't be notified about new messages.

Ah sorry, this might be because the term notification is misleading in the case of Blokada.
Activating `Notifications` in the case of Blokada shows a _permanent_, non-removable notification (at least in my case):

![blokada_notification](https://user-images.githubusercontent.com/372620/71811396-dec4cb00-3074-11ea-8072-36e79e78cff5.jpeg)

Messengers (like the mentioned ones in the first comment) don't have a _permanent_ notification. They only _show_ notifications when a message arrives but they are staying in background anyway.

**Comment by peterroth** [comment]
Tap on the Notifications line in Blokada, there cap on SETTINGS. Disable what you don't need. For example, I activated 'Keep alive' in Blokada, but disabled the notification for that in the system.
The result? The app runs in the background but I don't see a notification.

**Comment by alexanderadam** [comment]
Okay, I see, so you worked around the mentioned bug.

But then again it's pretty useless if Blokada is closed anyway if `Notification` is set to `No`, isn't it? :thinking:

**Comment by peterroth** [comment]
Why do you think it's a bug? The system will always have higher priority and the Notification options are there to increase Blokada's process priority to prevent being killed.

**Comment by alexanderadam** [comment]
Because of those two reasons:

1. Having the _option_ of showing Notifications does _not_ imply "_Hey, Blokada won't be functional if you don't activate it_". This like saying "_The car won't drive if you chose it in the colour 'blue'_". There's no way a regular user will see any semantic connection between the option and Blokada behaving malfunctioning.
2. As I mentioned before, other permanently running applications don't have the problem of having the need to show a permanent notification.

**Comment by peterroth** [comment]
For the 1st: no. The car won't drive if you don't pour some gasoline into it. Yes, the application will be killed if you don't enable notifications because the Android system is designed to kill/hibernate everything that has low priority and doesn't operate with notifications.
Other, permanently running applications also operate with notifications, either permanent or not. You can also swipe out the notifications of Blokada if you want to. I think the difference of a messaging app and Blokada is obvious from the point of view of notifications.
Those chat applications ask for permissions to show notifications at installation time. It's possible to enable 'Keep alive' by default, but then users need to hide that with the steps I described above, and that also leads us to the question of User Experience.

**Comment by alexanderadam** [comment]
> The car won't drive if you don't pour some gasoline into it.

This is exactly my point: why having an option "Remove all gasoline" in the settings then? :wink: 

> […] that also leads us to the question of User Experience.

I wholeheartedly agree! Why offering an _option_ if that will break the whole user experience?

> You can also swipe out the notifications of Blokada if you want to.

It's not working in my case. It's locked and if I try to swipe left/right it just shows the options of configuring notifications. I can't just swipe it away.

**Comment by peterroth** [comment]
So we are talking about a new default configuration proposition, and not about a bug ;) Asking for less permission ensures the user has better experience as we are talking about a security aware applications, furthermore, the current default configuration works on a big number of devices and Android versions.
However, it worth a revisit, for sure.

> It's not working in my case. It's locked and if I try to swipe left/right it just shows the options of configuring notifications. I can't just swipe it away.

I must check and validate it.

**Comment by peterroth** [comment]
I few versions ago, with 4.1 or around that, the Notification was enabled by default. Users complained why it is so, why they must have a constant notification, seeing that Blokada is active. This change was reverted but to the high number of user requests.
The proposal you have @alexanderadam was already delivered.

**Comment by alexanderadam** [comment]
I also mentioned that there should rather be a warning informs about possible malfunctioning of Blokada if someone want's to disable the notification ([last question of 1.](https://github.com/blokadaorg/blokada/issues/613#issuecomment-571095792)).

And proposal 2 is unanswered as well so far. :wink:

**Comment by cm-mc** [comment]
I noticed this before, too, but it seems 4.5.1 fixed it somehow, as I haven't experienced it since.

**Comment by alexanderadam** [comment]
So it turned out that this problem reappeared even when notifications were activated.

I tried using [personalDNSfilter](https://github.com/IngoZenz/personaldnsfilter) and [DNS66](https://github.com/julian-klode/dns66) in the last weeks and the problem doesn't appear on those alternatives so I simply switched.

Therefore I will simply close this issue because it seems that I was the only issue who ran into this issue anyway.

Thank you for your support!
